File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ans;
        int i = 0;
        
        while(numRows--){
            vector<int> v(i+1);
            v[0] = v[i] = 1;
            
            for(int j = 1; j < v.size() - 1 && i > 1; j++){
                v[j] = ans[i-1][j-1] + ans[i-1][j];
            }
            
            ans.push_back(v);
            i++;
        }
        
        return ans;
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.go
func generate(numRows int) [][]int { 
    res := make([][]int, numRows)
    
    for i := 0; i < numRows; i++ {
        level := make([]int, i+1)
        for j := 0; j <= i; j++ {
            if j == 0 || j == i {
                level[j] = 1
            } else {
                level[j] = res[i-1][j-1] + res[i-1][j]
            }
        }
        res[i] = level
    }
    
    return res
}